components/base/sensorcontrolled/spin.go

```diff
@@ -105,162 +52,88 @@ func (sb *sensorBase) stopSpinWithSensor(
 		timeOut = 10 * time.Second
 	}
 
-	sb.activeBackgroundWorkers.Add(1)
-	utils.ManagedGo(func() {
-		ticker := time.NewTicker(yawPollTime)
-		defer ticker.Stop()
+	for {
+		// check if we want to poll the sensor at all
+		// other API calls set this to false so that this for loop stops
 
-		for {
-			// check if we want to poll the sensor at all
-			// other API calls set this to false so that this for loop stops
-			if !sb.isPolling() {
-				ticker.Stop()
-			}
-
-			if err := ctx.Err(); err != nil {
-				ticker.Stop()
-				return
-			}
+		if !sb.isPolling() {
+			ticker.Stop()
+			sb.logger.CWarn(ctx, "Spin call interrupted by another running api")
+			return nil
+		}
 
-			select {
-			case <-ctx.Done():
-				ticker.Stop()
-				return
-			case <-ticker.C:
-				// imu readings are limited from 0 -> 360
-				currYaw, err := getCurrentYaw(sb.orientation)
-				if err != nil {
-					errCounter++
-					if errCounter > 100 {
-						sb.logger.CError(ctx, errors.Wrap(
-							err,
-							"imu sensor unreachable, 100 error counts when trying to read yaw angle, stopping base"))
-						if err = sb.Stop(ctx, nil); err != nil {
-							return
-						}
-					}
-					continue
-				}
-				errCounter = 0 // reset reading error count to zero if we are successfully reading again
+		if err := ctx.Err(); err != nil {
+			ticker.Stop()
+			return err
+		}
 
-				atTarget, overShot, minTravel := getTurnState(currYaw, startYaw, targetYaw, dir, angleDeg, errBound)
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-ticker.C:
+			orientation, err := sb.orientation.Orientation(ctx, nil)
+			if err != nil {
+				return err
+			}
 
-				if sensorDebug {
-					sb.logger.CDebugf(ctx, "minTravel %t, atTarget %t, overshot %t", minTravel, atTarget, overShot)
-					sb.logger.CDebugf(ctx, "angleDeg %.2f", angleDeg)
-					sb.logger.CDebugf(ctx,
-						"currYaw %.2f, startYaw %.2f, targetYaw %.2f",
-						currYaw, startYaw, targetYaw)
-				}
+			currYaw := rdkutils.RadToDeg(orientation.EulerAngles().Yaw)
+			angErr, prevMovedAng = getAngError(currYaw, prevAngle, prevMovedAng, angleDeg)
+			prevAngle = currYaw
 
-				// poll the sensor for the current error in angle
-				// check if we've overshot our target by the errTarget value
-				// check if we've travelled at all
-				if minTravel && (atTarget || overShot) {
-					if sensorDebug {
-						sb.logger.CDebugf(ctx,
-							"stopping base with errAngle:%.2f, overshot? %t",
-							math.Abs(targetYaw-currYaw), overShot)
-					}
+			if math.Abs(angErr) < boundCheckTarget {
+				return sb.Stop(ctx, nil)
+			}
```

Comment:
having the success check live within the same loop as the velocity commands made sense to me. Can refactor Spin to have multiple threads again if we prefer that.

Reply:
this makes sense to me